src/utils/config_validator.py

```diff
@@ -0,0 +1,217 @@
+""" verify that the user-provided config will execute the generation process
+correctly. Should a given argument be invalid, strings are appended to a list
+of errors.
+
+Error list to be returned, and used as a basis to feedback to the user that
+the configuration as-is is insufficient for successful operation.
+"""
+
+
+import sys
+from exceptions.config_error import ConfigError
+
+
+def validate(config):
+    """ Entry point. Build a list of errors based on a number of tests, when
+    returned can be used to feedback to the user what requires changing.
+
+    Parameters
+    ----------
+    config : dict
+        Parsed json of the user-input configuration
+
+    Raises
+    ------
+    ConfigError
+        Where an validation of the provided configuration has failed.
+        Raised only at the end of all checks such that all errors can be
+        reported at once.
+    """
+
+    # Temporarily remove traceback as all raised errors are from config #
+    sys.tracebacklimit = 0
+
+    domain_objects = config['domain_objects']
+    file_builders = config['file_builders']
+    shared_args = config['shared_args']
+
+    try:
+        errors = []
+        errors.append(validate_record_counts(domain_objects))
+        errors.append(validate_max_file_size(domain_objects))
+        errors.append(validate_output_file_extensions(file_builders,
+                                                      domain_objects))
+
+        errors.append(validate_pool_sizes_non_zero(shared_args))
+        errors.append(validate_job_size_non_zero(shared_args))
+
+        # Remove instances of None from error list
+        errors = [error for error in errors if error is not None]
+        # Remove instances of empty lists from error list
+        errors = [error for error in errors if error != []]
+        # Flatten list of lists to single list
+        errors = [error for sub_error in errors for error in sub_error]
+
+        if len(errors) is not 0:
+            raise ConfigError()
+
+    except ConfigError:
+        print("Issue(s) within Config:")
+        for error in errors:
+            print(error)
+        raise
+
+    else:
+        print("No issues within Config")
```

Comment:
Validate should return a ValidationResult.
ValidationResult should have a boolean success and an Array Errors[].
You can then test the result and get the error messages out.

The error should be thrown from App.py based on the result of the validation